rust/src/csv/reader.rs

```diff
@@ -70,21 +204,46 @@ impl Reader {
     /// Create a new CsvReader
     pub fn new(
         file: File,
-        schema: Arc<Schema>,
+        schema: Option<Arc<Schema>>,
         has_headers: bool,
+        delimiter: Option<u8>,
+        max_records: Option<usize>,
         batch_size: usize,
         projection: Option<Vec<usize>>,
     ) -> Self {
-        let csv_reader = csv::ReaderBuilder::new()
-            .has_headers(has_headers)
-            .from_reader(BufReader::new(file));
-
-        let record_iter = csv_reader.into_records();
-        Reader {
-            schema: schema.clone(),
-            projection,
-            record_iter,
-            batch_size,
+        match schema {
+            Some(schema) => {
+                let csv_reader = csv::ReaderBuilder::new()
+                    .delimiter(delimiter.unwrap_or(b','))
+                    .has_headers(has_headers)
+                    .from_reader(BufReader::new(file));
+                let record_iter = csv_reader.into_records();
+                Reader {
+                    schema: schema.clone(),
+                    projection,
+                    record_iter,
+                    batch_size,
+                }
+            }
+            None => {
```

Comment:
can we remove the shared code and only do `infger_schema` when `schema` is `None`?